src/morphology.rs

```diff
@@ -349,6 +349,185 @@ pub fn close_mut(image: &mut GrayImage, norm: Norm, k: u8) {
     erode_mut(image, norm, k);
 }
 
+/// A struct representing a mask used in morphological operations
+///
+/// the mask is represented by a list of the positions of its pixels
+/// relative to its center, with a maximum distance of 255
+/// along each axis.
+/// This means that in the most extreme case, the mask could have
+/// a size of 513 by 513 pixels.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Mask {
+    elements: Vec<(i16, i16)>,
+}
+
+impl Mask {
+    /// todo!()
+    pub fn square(radius: u8) -> Self {
+        Self {
+            elements: (-(radius as i16)..=(radius as i16))
+                .flat_map(|x| (-(radius as i16)..=(radius as i16)).map(move |y| (x, y)))
+                .collect(),
+        }
+    }
+
+    /// todo!()
+    pub fn diamond(radius: u8) -> Self {
+        Self {
+            elements: (-(radius as i16)..=(radius as i16))
+                .flat_map(|x| {
+                    ((x.abs() - radius as i16)..=(radius as i16 - x.abs())).map(move |y| (x, y))
+                })
+                .collect(),
+        }
+    }
+
+    /// todo!()
+    pub fn disk(radius: u8) -> Self {
+        let mut image = GrayImage::new(2 * radius as u32 + 1, 2 * radius as u32 + 1);
+        image.iter_mut().for_each(|p| *p = 0); // might not unnecessary
+        image.put_pixel(radius as u32, radius as u32, Luma::from([u8::MAX]));
+        dilate_mut(&mut image, Norm::L2, radius);
```

Comment:
There's a function for doing this directly: https://github.com/image-rs/imageproc/blob/master/src/drawing/conics.rs#L244

Reply:
thanks !

Reply:
i just realized that was a horrible way to do things and have rewritten this function in a much smarter way